core/src/main/kotlin/com/strumenta/kolasu/model/Node.kt

```diff
@@ -1,16 +1,25 @@
 package com.strumenta.kolasu.model
 
 import com.strumenta.kolasu.model.annotations.Annotation
-import com.strumenta.kolasu.model.observable.Observer
+import com.strumenta.kolasu.model.observable.AttributeChangedNotification
+import com.strumenta.kolasu.model.observable.NodeNotification
+import io.reactivex.rxjava3.core.ObservableSource
+import io.reactivex.rxjava3.core.Observer
+import io.reactivex.rxjava3.disposables.Disposable
 import java.io.Serializable
 import kotlin.reflect.KClass
 import kotlin.reflect.KMutableProperty
 import kotlin.reflect.KProperty1
 
+typealias NodeObserver = Observer<in NodeNotification<in Node>>
+
 /**
  * The Abstract Syntax Tree will be constituted by instances of Node.
  */
-open class Node() : Serializable {
+open class Node() : Serializable, ObservableSource<NodeNotification<in Node>>, Disposable {
+
+    @property:Internal
+    val observers: MutableList<NodeObserver> = mutableListOf()
```

Comment:
Also here I would use something like
```
val notifications = PublishSubject<NodeNotification<in Node>>.create()
```

Reply:
Implemented, please let me know if I did correctly.

I called it `changes` as in the other cases. Should I call it `notifications` here instead?

Reply:
Changes is fine I think